include/boost/geometry/algorithms/detail/buffer/buffer_inserter.hpp

```diff
@@ -177,14 +167,14 @@ struct buffer_range
         }
     }
 
-    static inline bool similar_direction(output_point_type const& p0,
+    // Returns true if collinear point p2 continues after p0 and p1.
+    // If it turns back (spike), it returns false.
+    static inline bool same_direction(output_point_type const& p0,
             output_point_type const& p1,
             output_point_type const& p2)
     {
-        typedef model::infinite_line<coordinate_type> line_type;
-        line_type const p = detail::make::make_infinite_line<coordinate_type>(p0, p1);
-        line_type const q = detail::make::make_infinite_line<coordinate_type>(p1, p2);
-        return arithmetic::similar_direction(p, q);
+        typedef typename cs_tag<output_point_type>::type cs_tag;
+        return direction_code<cs_tag>(p0, p1, p2) == 1;
```

Comment:
In geographic this'd need the earth model right?

Reply:
For this one I'm less sure (I mean: less than expand).
But I guess it's not necessary here either.

The spherical/geographical implementation is not made by me, and I didn't concentrate on it today, I just saw it was there and calling it makes the functionality less Cartesian dependent (than calling infinite_line).

Reply:
Ah right, this test has to be done after the collinearity check so we know the segments are collinear already and on the same ellipsoid.